src/Microsoft.DotNet.XHarness.CLI/Commands/WASI/Engine/WasiTestCommand.cs

```diff
@@ -65,8 +65,36 @@ protected override async Task<ExitCode> InvokeInternal(ILogger logger)
             logger.LogInformation($"Using wasm engine {Arguments.Engine.Value} from path {engineBinary}");
             await PrintVersionAsync(Arguments.Engine.Value.Value, engineBinary);
 
+            ServerURLs? serverURLs = null;
+            if (Arguments.WebServerMiddlewarePathsAndTypes.Value.Count > 0)
+            {
+                serverURLs = await WebServer.Start(
+                    Arguments,
+                    logger,
+                    cts.Token);
+                cts.CancelAfter(Arguments.Timeout);
```

Comment:
the way this is supposed to work is that you add the task into the `tasks` array and it'll trigger cancellation if the last task (the one with `Task.Delay(Arguments.Timeout)`) fires.

Reply:
There are multiple places where the `cts` is used here, it's not just one Task

Reply:
right, you don't need to add `cts.CancelAfter` here since that already happens later on